scripts/screens/cat_screens.py

```diff
@@ -2513,14 +2513,22 @@ def update_selected_cat(self):
                                                                                  object_id="#text_box_26_horizcenter_vertcenter_spacing_95",
                                                                                  manager=MANAGER)
 
-        if self.the_cat.status == "leader":
-            icon_path = "resources/images/leader_icon.png"
-        elif self.the_cat.status == "deputy":
-            icon_path = "resources/images/deputy_icon.png"
-        elif self.the_cat.status == "medicine cat":
-            icon_path = "resources/images/medic_icon.png"
-        elif self.the_cat.status == "medicine cat apprentice":
-            icon_path = "resources/images/medic_app_icon.png"
+        paths = {
+            "leader": "resources/images/leader_icon.png",
+            "deputy": "resources/images/deputy_icon.png",
+            "medicine cat": "resources/images/medic_icon.png",
+            "medicine cat apprentice": "resources/images/medic_app_icon.png",
+            "mediator": "resources/images/mediator_icon.png",
+            "mediator apprentice": "resources/images/mediator_app_icon.png",
+            "warrior": "resources/images/warrior_icon.png",
+            "apprentice": "resources/images/warrior_app_icon.png",
+            "kitten": "resources/images/kit_icon.png",
+            "newborn": "resources/images/kit_icon.png",
+            "elder": "resources/images/elder_icon.png",
+        }
+        
+        if self.the_cat.status in paths:
+            icon_path = paths[self.the_cat.status]
```

Comment:
For further improvement, I would suggest making this change:

```
        main_dir = "resources/images/"
        paths = {
            "leader": "leader_icon.png",
            "deputy": "deputy_icon.png",
            "medicine cat": "medic_icon.png",
            "medicine cat apprentice": "medic_app_icon.png",
            "mediator": "mediator_icon.png",
            "mediator apprentice": "mediator_app_icon.png",
            "warrior": "warrior_icon.png",
            "apprentice": "warrior_app_icon.png",
            "kitten": "kit_icon.png",
            "newborn": "kit_icon.png",
            "elder": "elder_icon.png",
        }

        if self.the_cat.status in paths:
            icon_path = os.path.join(main_dir,paths[self.the_cat.status])
```

Reply:
This is a good improvement - I will add it. Thanks!